server/lib/nl/common/utils.py

```diff
@@ -17,20 +17,23 @@
 import datetime
 import logging
 import random
+import re
 import time
 from typing import Any, Dict, List, Set
 
 import server.lib.fetch as fetch
 import server.lib.nl.common.constants as constants
 import server.lib.nl.common.counters as ctr
 import server.lib.nl.common.utterance as nl_uttr
+import server.lib.nl.detection.date
 from server.lib.nl.detection.types import ClassificationType
 import server.lib.nl.detection.types as types
 import server.lib.nl.fulfillment.utils as futils
 import shared.lib.constants as shared_constants
 
 # TODO: Consider tweaking/reducing this
 NUM_CHILD_PLACES_FOR_EXISTENCE = 15
+_FACET_OBS_PERIOD_RE = r'P(\d+)[Y|M|D|h|m|s]'
```

Comment:
if we only support YYYY-MM-DD date, do we still want `h|m|s`?

Reply:
we actually only support YYYY-MM so updated to just `Y|M`